dpnp/backend/extensions/sycl_ext/sum_mean.hpp

```diff
@@ -0,0 +1,216 @@
+//*****************************************************************************
+// Copyright (c) 2023, Intel Corporation
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+// - Redistributions of source code must retain the above copyright notice,
+//   this list of conditions and the following disclaimer.
+// - Redistributions in binary form must reproduce the above copyright notice,
+//   this list of conditions and the following disclaimer in the documentation
+//   and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+// THE POSSIBILITY OF SUCH DAMAGE.
+//*****************************************************************************
+
+#pragma once
+
+#include "dispatcher_utils.hpp"
+#include <CL/sycl.hpp>
+#include <tuple>
+
+#include "utils/memory_overlap.hpp"
+#include "utils/type_utils.hpp"
+
+namespace dpnp
+{
+namespace backend
+{
+namespace ext
+{
+namespace sycl_ext
+{
+
+namespace py = pybind11;
+namespace type_utils = dpctl::tensor::type_utils;
+
+bool check_limitations(const dpctl::tensor::usm_ndarray &in,
+                       const dpctl::tensor::usm_ndarray &out,
+                       bool throw_on_fail = false)
+{
+    if (not in.is_c_contiguous() or not out.is_c_contiguous()) {
+        if (throw_on_fail)
+            throw py::value_error(
+                "Input and output arrays must be c-contiguos");
+
+        return false;
+    }
+
+    auto device = in.get_queue().get_device();
+    auto local_mem_size = device.get_info<sycl::info::device::local_mem_size>();
+    size_t out_full_size = out.get_size() * out.get_elemsize();
+    if (out_full_size > local_mem_size) {
+        if (throw_on_fail)
+            throw py::value_error("Resulting array exceeds local memroy size" +
+                                  std::to_string(local_mem_size));
+
+        return false;
+    }
+
+    if (out.get_elemsize() == 64 and not device.has(sycl::aspect::atomic64)) {
```

Comment:
need to add a ckeck for `device.has(sycl::aspect::fp64)`

Reply:
looks not needed, since you have `validate_type_for_device()` calls

Reply:
It is verified here https://github.com/IntelPython/dpnp/pull/1479/files#diff-3ef38edc8347fa09d4d56d25def1932c8d15cf2d4cdc1b3306daa9836f38a729R161